tests/datatypes/test_str.py

```diff
@@ -529,6 +529,20 @@ def test_rindex(self):
                 print(err)
             """)
 
+    def test_replace(self):
+        self.assertCodeExecution("""
+            print('aaa'.replace('a', ''))
+            print('aaa'.replace('a', 's'))
+            print('aaa'.replace('a', 's',2))
+            print('aaa'.replace('a', 's',0))
+            print('abc'.replace('', 'M',5))
+            print('abc'.replace('', 'M',3))
+            print(''.replace('', 'M',10))
+            print(''.replace('', 'M',0))
+            print(''.replace('', 'M'))
+            print('abccabccabcc'.replace('abc', 'ab', 2))
```

Comment:
This is a good start, but I get the feeling there's a lot of tests missing here. Most of the tests involve single character substitutions on a single character match.

It would also be helpful if you annotated each test with a comment to describe what it was testing.